openadapt/privacy/providers/private_ai.py

```diff
@@ -0,0 +1,184 @@
+"""A Module for Private AI Scrubbing Provider."""
+
+from io import BytesIO
+from typing import List
+import base64
+import os
+
+from loguru import logger
+from PIL import Image
+import requests
+
+from openadapt import config
+from openadapt.privacy.base import Modality, ScrubbingProvider, TextScrubbingMixin
+from openadapt.privacy.providers import ScrubProvider
+
+
+class PrivateAIScrubbingProvider(
+    ScrubProvider, ScrubbingProvider, TextScrubbingMixin
+):  # pylint: disable=abstract-method
+    """A Class for Private AI Scrubbing Provider."""
+
+    name: str = ScrubProvider.PRIVATE_AI
+    capabilities: List[Modality] = [Modality.TEXT, Modality.PIL_IMAGE, Modality.PDF]
+
+    def scrub_text(self, text: str, is_separated: bool = False) -> str:
+        """Scrub the text of all PII/PHI.
+
+        Args:
+            text (str): Text to be scrubbed
+            is_separated (bool): Whether the text is separated with special characters
+
+        Returns:
+            str: Scrubbed text
+        """
+        url = "https://api.private-ai.com/deid/v3/process/text"
+
+        payload = {
+            "text": [text],
+            "link_batch": False,
+            "entity_detection": {
+                "accuracy": "high",
+                "return_entity": True,
+            },
+            "processed_text": {
+                "type": "MARKER",
+                "pattern": "[UNIQUE_NUMBERED_ENTITY_TYPE]",
+            },
+        }
+
+        headers = {
+            "Content-Type": "application/json",
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(url, json=payload, headers=headers)
+        if response is None:
+            raise ValueError("Private AI request returned None")
+
+        data = response.json()
+        logger.debug(data)
+        if type(data) is dict and "detail" in data.keys():
+            raise ValueError(data.get("detail"))
+
+        redacted_text = data[0].get("processed_text")
+        logger.debug(redacted_text)
+
+        return redacted_text
+
+    def scrub_image(
+        self,
+        image: Image,
+        fill_color: int = config.SCRUB_FILL_COLOR,  # pylint: disable=no-member
+    ) -> Image:
+        """Scrub the image of all PII/PHI.
+
+        Args:
+            image (Image): A PIL.Image object to be scrubbed
+            fill_color (int): The color used to fill the redacted regions(BGR).
+
+        Returns:
+            Image: The scrubbed image with PII and PHI removed.
+        """
+        url = "https://api.private-ai.com/deid/v3/process/files/base64"
+        file_dir = "assets/"
+        file_name = "temp_image_to_scrub.png"
+
+        # save file as "temp_image_to_scrub.png in assets/
+        temp_image_path = os.path.join(file_dir, file_name)
+        image.save(temp_image_path)
+
+        file_type = "image/png"
+
+        # Read from file
+        with open(temp_image_path, "rb") as b64_file:
```

Comment:
Can we please use `io.BytesIO` instead of writing to a file? https://docs.python.org/3/library/io.html